eng/common/TestResources/New-TestResources.ps1

```diff
@@ -279,46 +279,13 @@ try {
         if (!$TenantId) {
             $TenantId = $context.Subscription.TenantId
         }
-
-        # If no test application ID is specified during an interactive session, create a new service principal.
-        if (!$TestApplicationId) {
-
-            # Cache the created service principal in this session for frequent reuse.
-            $servicePrincipal = if ($AzureTestPrincipal -and (Get-AzADServicePrincipal -ApplicationId $AzureTestPrincipal.ApplicationId) -and $AzureTestSubscription -eq $SubscriptionId) {
-                Log "TestApplicationId was not specified; loading cached service principal '$($AzureTestPrincipal.ApplicationId)'"
-                $AzureTestPrincipal
-            } else {
-                Log "TestApplicationId was not specified; creating a new service principal in subscription '$SubscriptionId'"
-                $suffix = (New-Guid).ToString('n').Substring(0, 4)
-                $global:AzureTestPrincipal = New-AzADServicePrincipal -Role Owner -Scope "/subscriptions/$SubscriptionId" -DisplayName "test-resources-$($baseName)$suffix.microsoft.com"
-                $global:AzureTestSubscription = $SubscriptionId
-
-                Log "Created service principal '$($AzureTestPrincipal.ApplicationId)'"
-                $AzureTestPrincipal
-            }
-
-            $TestApplicationId = $servicePrincipal.ApplicationId
-            $TestApplicationSecret = (ConvertFrom-SecureString $servicePrincipal.Secret -AsPlainText);
-
-            # Make sure pre- and post-scripts are passed formerly required arguments.
-            $PSBoundParameters['TestApplicationId'] = $TestApplicationId
-            $PSBoundParameters['TestApplicationSecret'] = $TestApplicationSecret
-        }
-
-        if (!$ProvisionerApplicationId) {
-            $ProvisionerApplicationId = $TestApplicationId
-            $ProvisionerApplicationSecret = $TestApplicationSecret
-            $TenantId = $context.Tenant.Id
-        }
     }
 
-    # Log in as and run pre- and post-scripts as the provisioner service principal.
+    # If a provisioner service principal was provided, log into it to perform the pre- and post-scripts and deployments.
     if ($ProvisionerApplicationId) {
         $null = Disable-AzContextAutosave -Scope Process
 
         Log "Logging into service principal '$ProvisionerApplicationId'."
-        Write-Warning 'Logging into service principal may fail until the principal is fully propagated.'
```

Comment:
This needs to stay. This was specifically added because of numerous people complaining they often see 1 or 2 attempts.

Reply:
I removed this because we're no longer misusing the Test Application as a provisioner; the only reason that we'd see failures is if the provisioner was created immediately prior to calling the script.    

I'm not against adding it back, I just don't know that it makes sense in the current context.